pubtools/_quay/push_docker.py

```diff
@@ -0,0 +1,194 @@
+        push_items (<list-of-push-items>)
+            List of push items containing metadata
+        signing_key (str)
+            Signing key which should be used to sign pushed manifests
+        hub (xmlrpc.client.ServerProxy)
+            Pubhub XMLRPC proxy instance
+        task_id (int)
+            Pub task id
+        target_name (str)
+            Pub target name
+        target_settings (dict)
+            Target settings
+    """
+    log_push_items(signing_key, items=push_items)
+    shared_data = {}
+    logger = Logger()
+    common_external_res = {
+        "push_items": push_items,
+        "log_info": logger.log_info,
+        "log_error": logger.log_error,
+        "log_warning": logger.log_warning,
+        "log_debug": logger.log_debug,
+        "hub": hub,
+        "task_id": task_id,
+        "target_name": target_name,
+    }
+
+    stepper = Stepper(shared_data)
+    stepper.add_step(
+        StepSanitizeContainerPushItems(
+            "1", (), {}, shared_data, external_resources=common_external_res
+        )
+    )
+    stepper.add_step(
+        StepSanitizeRepositories(
+            "1",
+            ("StepSanitizeContainerPushItems:1",),
+            {},
+            shared_data,
+            external_resources=common_external_res,
+        )
+    )
+    stepper.add_step(
+        StepBuildBackupMapping(
+            "1",
+            ("StepSanitizeContainerPushItems:1",),
+            {},
+            shared_data,
+            external_resources=common_external_res,
+        )
+    )
+    stepper.add_step(
+        StepPushContainerImgs(
+            "1",
+            ("StepSanitizeContainerPushItems:1", target_settings),
+            {},
+            shared_data,
+            external_resources=common_external_res,
+        )
+    )
+    stepper.add_step(
+        StepMergeManifestList(
+            "1",
+            ("StepSanitizeContainerPushItems:1", target_settings),
+            {},
+            shared_data,
+            external_resources=common_external_res,
+        )
+    )
+    stepper.add_step(
+        StepSignContainers(
+            "1",
+            ("StepSanitizeContainerPushItems:1", target_settings),
+            {
+                "docker_container_signing_enabled": target_settings[
+                    "docker_settings"
+                ].get("docker_container_signing_enabled", False)
+            },
+            shared_data,
+            external_resources=common_external_res,
+        )
+    )
+    stepper.add_step(
+        StepSanitizeOperatorPushItems(
+            "1",
+            (),
+            {"auto_upload_operators": target_settings.get("auto_upload_operators")},
+            shared_data,
+            external_resources=common_external_res,
+        )
+    )
+    stepper.add_step(
+        StepPushOperators(
+            "1",
+            ("StepSanitizeOperatorPushItems:1", target_settings),
+            {
+                "docker_reference_registry": target_settings.get(
```

Comment:
docker_reference_registry is in target_settings["docker_settings"].

Reply:
Yes, that's for pulp targets, but I think for quay it doesn't make sense separate this in extra dictionary.

Reply:
How about docker_container_signing_enabled?

Reply:
good point, I'll change it to container_signing_enabled

Reply:
But why it needs to be in target_settings["docker_settings"] not target_settings? In Line 123.